src/pages/_layout.svelte

```diff
@@ -66,11 +66,33 @@ $: menuItems = [
     hide: inAdminRole || !policyId,
   },
 ]
+const isCustomerOnOwnPolicy = (policyId: string) => policyId === $selectedPolicyId
 
-const goToPolicyAsCustomer = (event: CustomEvent) => $goto(routes.policyHome(event.detail))
-const goToAdminHome = () => $goto(routes.ADMIN_HOME)
+const gotoPath = (policyId: string, claimOrItemIdObj = {}) =>
+  $goto($url($route.path, { policyId, ...claimOrItemIdObj }))
+
+const goToCustomerView = (event: CustomEvent) => {
+  if ($params.policyId && !$params.claimId && !$params.itemId) {
+    gotoPath(event.detail)
+  } else if ($params.policyId && ($params.claimId || $params.itemId) && isCustomerOnOwnPolicy(event.detail)) {
+    const claimOrItemIdObj = $params.claimId ? { claimId: $params.claimId } : { itemId: $params.itemId }
+    gotoPath(event.detail, claimOrItemIdObj)
+  } else {
+    $goto(routes.policyHome(event.detail))
+  }
+}
+const goToAdminView = (event: CustomEvent) => {
+  if ($params.policyId && ($params.claimId || $params.itemId)) {
+    const claimOrItemIdObj = $params.claimId ? { claimId: $params.claimId } : { itemId: $params.itemId }
+    gotoPath(event.detail.policyId, claimOrItemIdObj)
+  } else if ($params.policyId) {
+    gotoPath(event.detail.policyId)
+  } else {
+    $goto(routes.ADMIN_HOME)
+  }
```

Comment:
It seems like it might simplify the code some here to check for URL params more individually:
```suggestion
  if ($params.policyId) {
    if ($params.claimId) {
      gotoPath(event.detail.policyId, { claimId: $params.claimId })
    } else if ($params.itemId) {
      gotoPath(event.detail.policyId, { itemId: $params.itemId })
    } else {
      gotoPath(event.detail.policyId)
    }
  } else {
    $goto(routes.ADMIN_HOME)
  }
```